sound/soc/sof/intel/Kconfig

```diff
@@ -85,10 +85,18 @@ config SND_SOC_SOF_BAYTRAIL
 
 config SND_SOC_SOF_BROADWELL_SUPPORT
 	bool "SOF support for Broadwell"
+	depends on SND_SOC_INTEL_HASWELL=n
 	help
 	  This adds support for Sound Open Firmware for Intel(R) platforms
 	  using the Broadwell processors.
-	  Say Y if you have such a device.
+	  This option is mutually exclusive with the Haswell/Broadwell legacy
+	  driver. If you want to enable SOF on Broadwell you need to deselect
+	  the legacy driver first.
```

Comment:
that's how mutual exclusion based on 'depends' work. I have no ability to change this.
All I could do is add text in the help to say that the legacy option gates SOF.

Reply:
My suggestion would be to move the second sentence here to the help text for `SND_SOC_INTEL_HASWELL` (rewording as needed).

As it stands now, if the user is able to read the second sentence at all, they can basically ignore, it because the legacy driver is already deselected.

If this doesn't follow the pattern used elsewhere in Kconfig files though (which seems to vary by subsystem), then please disregard.

Reply:
I'll add an explanation on the legacy side as well.

Reply:
Is not that change against the goal I meant with the sound/hda/intel-dsp-config.c ? For distros, we need to have all drivers available and select the correct drivers at runtime (automatically and by the user config). I believe that there should not be any "depends on SND_SOC_INTEL_HASWELL=n" config dependencies at all.

Reply:
I'm not sure which question is being asked here.
As of the time this was merged, sound/hda/intel-dsp-config.c is [not sufficient](https://bugzilla.kernel.org/show_bug.cgi?id=204237#c20) to prevent kernel bug 204237 from happening.

Reply:
Intel-dsp-config.c should resolve the issue for the PCI probe, but it seems that we need to do this for ACPI probe, too. @plbossart, I'm right? The target is to be able to compile both SOF / SST drivers for distros.

Reply:
Yes, I don't know how to do a dynamic detection for Broadwell and Baytrail. SOF is a replacement but it's not ready for prime time just yet.